src/pages/TeamOverview/__tests__/TeamOverview.test.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import {render, screen, waitFor} from '@testing-library/react';
+import {TeamOverviewType, UserDataType} from 'types/types';
+import * as API from '../../../services/api';
+import TeamOverview from '../TeamOverview';
+
+jest.mock('react-router-dom', () => ({
+    useLocation: () => ({
+        state: {
+            id: '1',
+            name: 'Some Team',
+        },
+    }),
+    useNavigate: () => jest.fn(),
+    useParams: () => ({
+        teamId: '1',
+    }),
+}));
+
+const getUserData = (id) => {
+    const userData: UserDataType = {
+        id,
+        firstName: 'userData',
+        lastName: 'userData',
+        displayName: 'userData',
+        location: '',
+        avatar: '',
+    };
+
+    return userData;
+};
+
+describe('TeamOverview', () => {
+    beforeAll(() => {
+        jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+        jest.clearAllTimers();
+    });
+
+    afterAll(() => {
+        jest.useRealTimers();
+    });
+
+    it('should render team overview users', async () => {
+        const teamOverview: TeamOverviewType = {
+            id: '1',
+            teamLeadId: '2',
+            teamMemberIds: ['3', '4', '5'],
+        };
+        const teamLeadData = getUserData('2');
+        const teamMembersData = [
+            getUserData('3'),
+            getUserData('4'),
+            getUserData('5'),
+        ];
+        jest.spyOn(API, 'getTeamOverview').mockImplementationOnce(() => Promise.resolve(teamOverview));
+        jest.spyOn(API, 'getUserData')
```

Comment:
to override the `getUserData` , that is called for the `teamLeader` and all teamMembers, resulting in 4 users total